tools/topology/topology2/include/dais/alh.conf

```diff
@@ -0,0 +1,86 @@
+#
+# ALH DAI class definition. All attributes defined herein are namespaced
+# by alsatplg to "Object.Dai.ALH.N.attribute_name"
+#
+# Usage: ALH DAI objects can be instantiated as:
+#
+# Object.Dai.ALH."N" {
+# 	direction		"capture" # playback or capture
+# 	dai_name		"SDW0-Capture"
+# 	id 			2
+# 	Object.hw_config."2" {
+# 	}
+# }
+#
+# where N is the unique instance number for the ALH object within the same alsaconf node.
+
+# ALH port definition
+Class.Dai."ALH" {
+
+	#
+	# Argument used to construct DAI widget
+	#
+	# Playback DAI Index
+	DefineAttribute."dai_index" {
+		token_ref	"sof_tkn_dai.word"
+	}
+
+	DefineAttribute."direction" {
+		type "string"
+		constraints {
+			!valid_values [
+				"playback"
+				"capture"
+				"duplex"
+			]
+		}
+	}
+
+	DefineAttribute."dai_type" {
+		type	"string"
+		token_ref	"sof_tkn_dai.string"
+	}
+
+	DefineAttribute."default_hw_config_id" {}
+
+	DefineAttribute."name" {
+		type	"string"
+	}
+
+	DefineAttribute."type" {
+		token_ref	"sof_dai_tokens.string"
+	}
+
+	DefineAttribute."rate" {
+		token_ref	"sof_tkn_alh.word"
+	}
+
+	DefineAttribute."ch" {
+		token_ref	"sof_tkn_alh.word"
+	}
+
+	# Backend DAI Link ID matching with the machine driver
+	DefineAttribute."id" {}
+	DefineAttribute."default_hw_config_id" {}
+	DefineAttribute."dai_name" {}
+
+
+	attributes {
+		constructor [
```

Comment:
For `constructor/mandatory/immutable` and other array attributes, please add `!` to avoid the problem when this conf is include by other conf directly or indirectly more than once.  check this issue: https://github.com/alsa-project/alsa-lib/issues/151. Other part is good for me.

Reply:
Done